presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -153,6 +155,7 @@ public class IcebergHiveMetadata
     private final DateTimeZone timeZone = DateTimeZone.forTimeZone(TimeZone.getTimeZone(ZoneId.of(TimeZone.getDefault().getID())));
 
     private final FilterStatsCalculatorService filterStatsCalculatorService;
+    private final ConcurrentMap<SchemaTableName, org.apache.iceberg.Table> icebergTable = new ConcurrentHashMap<>();
```

Comment:
Why not use one of the guava `Cache` implementations?

Reply:
In the same vein as Zac's comment, but more of a NIT: I think it would make sense to have this variable name either be plural, such as `icebergTables`, or indicate that it's a cache, such as `tableMetadataCache` (and same in `IcebergNativeMetadata`)

Reply:
+1 to the rename.  I think it's fine to keep it as a map, since it's a per-query map and should not have an expiration.

Reply:
@ZacBlanco I thought about using guava `Cache` implementation for this but that would have other variables to be considered carefully like cache invalidation and table versions etc so I thought we could introduce Cache as next advancement if required. And maintaining Query level table metadata provides us comparable optimization as using Cache and doesn't have other variables to maintain.